### PR TITLE
check for dovecot supported password schemes

### PR DESCRIPTION
**Pull request by Toniob**
call doveadm pw -l to list password schemes
only show supported password schemes

### CODE REVIEW

File: modoboa/core/password_hashers/__init__.py [hunk 1]
```diff
@@ -29,3 +34,26 @@ def get_password_hasher(scheme):
...
+    supported_schemes = ["{PLAIN}"]
+    dpath = None
+    code, output = exec_cmd("which doveadm")
+    if not code:
+        dpath = force_text(output).strip()
+    else:
+        known_paths = ("/usr/bin/doveadm", "/usr/local/bin/doveadm")
+        for fpath in known_paths:
+            if os.path.isfile(fpath) and os.access(fpath, os.X_OK):
+                dpath = fpath
```
Comment (tonioo) [inline modoboa/core/password_hashers/__init__.py]: You can add a break once you've found a match.

File: modoboa/core/password_hashers/__init__.py [hunk 2]
```diff
@@ -29,3 +34,26 @@ def get_password_hasher(scheme):
...
+    if dpath:
+        try:
+            code, schemes = exec_cmd("%s pw -l" % dpath)
```
Comment (tonioo) [inline modoboa/core/password_hashers/__init__.py]: no .format() here?

File: modoboa/core/password_hashers/__init__.py [hunk 3]
```diff
@@ -29,3 +34,26 @@ def get_password_hasher(scheme):
...
+
+def get_dovecot_schemes():
+    """Return schemes supported by dovecot"""
+    supported_schemes = ["{PLAIN}"]
+    dpath = None
+    code, output = exec_cmd("which doveadm")
+    if not code:
+        dpath = force_text(output).strip()
+    else:
+        known_paths = ("/usr/bin/doveadm", "/usr/local/bin/doveadm")
```
Comment (tonioo) [inline modoboa/core/password_hashers/__init__.py]: I think there is another place in the code where we look for doveadm path. I guess a new function would be welcome? (maybe in lib/sysutils.py)

File: modoboa/core/password_hashers/advanced.py [hunk 4]
```diff
@@ -49,6 +52,9 @@ class MD5CRYPTHasher(PasswordHasher):
 
     This scheme can't be considered as secure anymore.
     """
+    name = "md5crypt"
+    label = "md5crypt"
```
Comment (tonioo) [inline modoboa/core/password_hashers/advanced.py]: Two properties in the base class would be better. The scheme name can be guessed from the class name and we could add a weak flag too.

**Comment by Toniob** [comment]
closes #1584

**Pull request merged**